types/convert.go

```diff
@@ -369,13 +369,20 @@ func getValidIntPrefix(sc *stmtctx.StatementContext, str string) (string, error)
 	return floatStrToIntStr(sc, floatPrefix, str)
 }
 
-// roundIntStr is to round int string base on the number following dot.
+// roundIntStr is to round a **valid int string** base on the number following dot.
 func roundIntStr(numNextDot byte, intStr string) string {
 	if numNextDot < '5' {
 		return intStr
 	}
 	retStr := []byte(intStr)
 	for i := len(intStr) - 1; i >= 0; i-- {
+		if !isDigit(intStr[i]) {
```

Comment:
here `i` must be 0 ?

Reply:
yes, but this a valid int string, so we needn't to check whether it is 0.

Reply:
If `i` must be 0, can we move `if !isDigit(intStr[i])` logic to outside the `for loop`?

Reply:
done